sdk/resourcemanager/Azure.ResourceManager/docs/MigrationGuide.md

```diff
@@ -0,0 +1,300 @@
+#### Old
+```C#
+ServiceClientCredentials credentials = getMyCredentials(); 
+ResourceManagementClient resourcesClient = new ResourceManagementClient(credentials);
+
+string rgName = "QuickStartRG";
+string location = "WestUS2";
+resourcesClient.ResourceGroups.CreateOrUpdate(
+    rgName,
+    new ResourceGroup
+    {
+        Location = location,
+        Tags = new Dictionary<string, string>() { { rgName, DateTime.UtcNow.ToString("u") } }
+    });
+```
+#### New
+```C#
+Subscription subscription = armClient.DefaultSubscription;
+ResourceGroupContainer rgContainer = subscription.GetResourceGroups();
+
+Location location = Location.WestUS2;
+string rgName = "QuickStartRG";
+
+ResourceGroupData rgData = new ResourceGroupData(location);
+ResourceGroup resourceGroup = await rgContainer.CreateOrUpdateAsync(rgName, rgData);
+```
+The main difference is that the previous libraries represent all operations as flat, while the new preview libraries respresents the hierarchy of resources. In that way, you can use a `subscriptionContainer` to manage the resources in a particular subscription. In this example, a `resourceGroupContainer` is used to manage the resources in a particular resource group. In the example above, a new resource group is created from a resourceGroupContainer. With that `ResourceGroup` you will be able to get the resource containers to manage all the resources that will be inside it, as it is shown in the next part of this guide.
+
+Another thing is that the new preview SDK provides some common classes to represent commonly-used constructs, like `Location`, and allow you to use them directly throughout the APIs, again, making it easier to discover how to properly configure resources.
+
+### Create an Availability Set
+#### Old
+```C#
+AvailabilitySet inputAvailabilitySet = new AvailabilitySet
+{
+    Location = location,
+    Tags = new Dictionary<string, string>()
+    {
+        { "RG", "rg" },
+        { "testTag", "1" }
+    },
+    PlatformFaultDomainCount = 1,
+    PlatformUpdateDomainCount = 1,
+    Sku = new CM.Sku
+    {
+        Name = AvailabilitySetSkuTypes.Aligned 
+    }
+};
+string aSetName = "quickstartvm_aSet";
+
+AvailabilitySet asCreateOrUpdateResponse = computeClient.AvailabilitySets.CreateOrUpdate(rgName,aSetName,inputAvailabilitySet);
+string aSetID = $"/subscriptions/{computeClient.SubscriptionId}/resourceGroups/{rgName}/providers/Microsoft.Compute/availabilitySets/{aSetName}";
+```
+#### New
+```C#
+string vmName = "quickstartvm";
+AvailabilitySetData aSetData = new AvailabilitySetData(location);
+AvailabilitySet aset = (await resourceGroup.GetAvailabilitySets().CreateOrUpdateAsync(vmName + "_aSet", aSetData)).Value;
+string asetId = aset.Id;
+```
+
+Parameters can be specified via the `AvailabilitySetData` object, in here, the basic default only requires the location. The availability set is created using  the `GetAvailabilitySets()` extension method instead of using another client. 
```

Comment:
```suggestion
Parameters can be specified via the `AvailabilitySetData` object, in here, the basic default only requires the location. The availability set is created using  the AvailabilitySetsContainer returned from the `GetAvailabilitySets()` extension method instead of using another client. 
```